src/components/autocomplete/js/autocompleteController.js

```diff
@@ -274,9 +274,11 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
         $scope.searchText = val;
         handleSelectedItemChange(selectedItem, previousSelectedItem);
       });
+    } else {
```

Comment:
This clears the search text if the underlying model is cleared. But yeah you're right, I'll move this to a separate issue/pr.